airbyte-integrations/connectors/source-klarna/source_klarna/source.py

```diff
@@ -0,0 +1,123 @@
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_json = response.json()
+        if 'pagination' in response_json:
+            if 'next' in response_json['pagination']:
+                parsed_url = urlparse(response_json['pagination']['next'])
+                query_params = parse_qs(parsed_url.query)
+                # noinspection PyTypeChecker
+                return query_params
+        else:
+            return None
+
+    def request_params(
+            self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        if next_page_token:
+            return dict(next_page_token)
+        else:
+            return {"offset": 0, "size": self.page_size}
+
+
+class Payouts(KlarnaStream):
+    """
+    Payouts read from Klarna Settlements API https://developers.klarna.com/api/?json#settlements-api
+    """
+    primary_key = "payout_date"  # TODO verify
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "/settlements/v1/payouts"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        payouts = response.json().get("payouts", [])
+        yield from payouts
+
+
+class Transactions(KlarnaStream):
+    """
+    Transactions read from Klarna Settlements API https://developers.klarna.com/api/?json#settlements-api
+    """
+    primary_key = "capture_id"  # TODO verify
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "/settlements/v1/transactions"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        transactions = response.json().get("transactions", [])
+        yield from transactions
+
+
+# Source
+class SourceKlarna(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        :param config:  the user-input config object conforming to the connector's spec.yaml
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        try:
+            auth = BasicHttpAuthenticator(username=config['username'], password=config['password'])
+            conn_test_stream = Transactions(authenticator=auth, **config)
+            conn_test_stream.page_size = 1
+            conn_test_stream.next_page_token = lambda x: None
```

Comment:
Why this?

Reply:
I just want to do one http call with size 1 to decrease time needed for checks. It is not possible to pass any argument to the `read_records` that would allow me to do so then just monkey_patching method for `next_page_token`